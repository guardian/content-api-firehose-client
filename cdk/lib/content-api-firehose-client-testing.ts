import type { GuStackProps } from '@guardian/cdk/lib/constructs/core';
import { GuStack } from '@guardian/cdk/lib/constructs/core';
import { GuGithubActionsRole } from '@guardian/cdk/lib/constructs/iam';
import { GuAllowPolicy } from '@guardian/cdk/lib/constructs/iam/policies/base-policy';
import type { App } from 'aws-cdk-lib';

export class ContentApiFirehoseClientTesting extends GuStack {
	constructor(scope: App, id: string, props: GuStackProps) {
		super(scope, id, props);
		new GuGithubActionsRole(this, {
			policies: [
				new GuAllowPolicy(this, 'kinesis-stream-access', {
					actions: ['kinesis:DescribeStreamSummary'],
					resources: [
						`arn:aws:kinesis:${this.region}:${this.account}:stream/content-api-firehose-v2-CODE`,
					],
				}),
			],
			condition: {
				githubOrganisation: 'guardian',
				repositories: 'content-api-firehose-client:*',
			},
		});
	}
}
