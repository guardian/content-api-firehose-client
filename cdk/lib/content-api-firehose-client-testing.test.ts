import { App } from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import { ContentApiFirehoseClientTesting } from "./content-api-firehose-client-testing";

describe("The ContentApiFirehoseClientTesting stack", () => {
  it("matches the snapshot", () => {
    const app = new App();
    const stack = new ContentApiFirehoseClientTesting(app, "ContentApiFirehoseClientTesting", { stack: "content-api-firehose-client", stage: "TEST" });
    const template = Template.fromStack(stack);
    expect(template.toJSON()).toMatchSnapshot();
  });
});
