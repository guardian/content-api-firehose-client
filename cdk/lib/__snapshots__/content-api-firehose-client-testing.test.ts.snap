// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ContentApiFirehoseClientTesting stack matches the snapshot 1`] = `
{
  "Metadata": {
    "gu:cdk:constructs": [
      "GuAllowPolicy",
      "GuGithubActionsRole",
    ],
    "gu:cdk:version": "TEST",
  },
  "Outputs": {
    "GithubActionsRoleGithubActionsRoleArnC13D9654": {
      "Value": {
        "Fn::GetAtt": [
          "GithubActionsRoleF5CC769F",
          "Arn",
        ],
      },
    },
  },
  "Resources": {
    "GithubActionsRoleF5CC769F": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRoleWithWebIdentity",
              "Condition": {
                "StringLike": {
                  "token.actions.githubusercontent.com:sub": "repo:guardian/content-api-firehose-client:*",
                },
              },
              "Effect": "Allow",
              "Principal": {
                "Federated": {
                  "Fn::Join": [
                    "",
                    [
                      "arn:aws:iam::",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":oidc-provider/token.actions.githubusercontent.com",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/content-api-firehose-client",
          },
          {
            "Key": "Stack",
            "Value": "content-api-firehose-client",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "kinesisstreamaccessFE21B9E2": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "kinesis:DescribeStreamSummary",
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:kinesis:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":stream/content-api-firehose-v2-CODE",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "kinesisstreamaccessFE21B9E2",
        "Roles": [
          {
            "Ref": "GithubActionsRoleF5CC769F",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
}
`;
